crates/testing/src/main.rs

```diff
@@ -0,0 +1,27 @@
+use std::sync::Arc;
+
+use clap::Parser as _;
+use testing::slt::{
+    cli::Cli,
+    hooks::{AllTestsHook, SshTunnelHook},
+    runner::SltRunner,
+    tests::{PgBinaryEncoding, SshKeysTest},
+};
+
+pub fn main() -> anyhow::Result<()> {
+    let cli = Cli::parse();
+    let test_dir = cli
+        .test_dir
+        .unwrap_or_else(|| std::env::current_dir().unwrap());
+
+    SltRunner::new()
+        .test_files_dir(test_dir)?
+        // Rust tests
+        .test("sqllogictests/ssh_keys", Box::new(SshKeysTest))?
+        .test("pgproto/binary_encoding", Box::new(PgBinaryEncoding))?
+        // Add hooks
+        .hook("*", Arc::new(AllTestsHook))?
+        // SSH Tunnels hook
+        .hook("*/tunnels/ssh", Arc::new(SshTunnelHook))?
+        .run()
+}
```

Comment:
we could shave another minute off our builds if we want to include this runner into the glaredb binary directly.

Reply:
I was initially viscerally opposed to this idea, but as I look at this code and also think about it a bit more, I kind of like it. 

Potential caveats: 
- I think the reset of the team should definitely sign off on this. 😁 😓
- I think we should hide it from any help menus 
- I think there should be a pre-flight log warning message that says "the slt test runner is not supported and does not have the same stability guarantees as the rest of glaredb." 

Otherwise, I think maybe it should be in everything, including the release build, because:
- I'm a little afraid of having the dev build drift/vary that much from the test build. 
- making it super easy for users to run/validate a simple reproduction case using slt on their hardware/systems seems like it might be useful.

Reply:
> Otherwise, I think maybe it should be in everything, including the release build,

I think this would be a pretty good candidate for a feature flag, as it's something thats unlikely to change very often & has a very clear boundary on when/why you'd want to enable the feature. But I'm open for suggestions.

Reply:
Would including the the runner in glaredb _just_ be for build times, or would there be some other benefits as well.

Some other questions:
- Why is this binary not built during the "build" step? Are we just missing a cargo profile or something?

Reply:
> Would including the the runner in glaredb just be for build times, or would there be some other benefits as well.

I think build times are one, but not just for CI, but also dev workflow. I often find myself in the workflow of: 
run slts
run local 
execute some queries for debugging
make small changes
repeat

which is not great because `run local` and `slt` trigger different build paths so it usually slows down development quite a bit going back and forth between the two. if slt was part of the main binary, it'd speed up this cycle quite a bit. 

> Why is this binary not built during the "build" step? Are we just missing a cargo profile or something?

We could throw it into the "build" step, but that wouldn't really add any benefits.  Since it's a separate crate/bin, we need to compile it separately.